src/main/java/seedu/address/logic/commands/UntagCommand.java

```diff
@@ -84,14 +102,38 @@ public boolean equals(Object other) {
         }
 
         UntagCommand otherTagCommand = (UntagCommand) other;
-        return targetIndex.equals(otherTagCommand.targetIndex) && tag.equals(otherTagCommand.tag);
+        return targetIndexes.equals(otherTagCommand.targetIndexes) && tags.equals(otherTagCommand.tags);
     }
 
     @Override
     public String toString() {
         return new ToStringBuilder(this)
-                .add("targetIndex", targetIndex)
-                .add("tag", tag)
+                .add("targetIndexes", targetIndexes)
+                .add("tag", tags)
                 .toString();
     }
+
+    private Person setPerson(Model model, Person personToUntag, Set<Tag> newTags) {
+        Person updatedPerson = new Person(personToUntag.getName(), personToUntag.getPhone(),
+                personToUntag.getEmail(), personToUntag.getRsvpStatus(), newTags);
+        model.setPerson(personToUntag, updatedPerson);
+        return updatedPerson;
+    }
```

Comment:
I'm not sure if this entire method should be placed in the ModelManager class and called directly instead of having a middleman here. Passing the model in as an argument seems counterintuitive since the main operation acts directly on the model (model.setPerson). Let me know what you think!

Reply:
I agree that the main operation acts directly on the model, however since the execute method is long to handle different cases, I though it would be better to extract out this chunk to increase readability.